lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -405,6 +409,14 @@ export abstract class CacheManager implements ICacheManager {
             if (!!target && !this.matchTarget(entity, target)) {
                 return;
             }
+            
+            // If request OR cached entity has requested Claims Hash, check if they match
+            if (requestedClaimsHash || entity.requestedClaimsHash) {
+                // Don't match if either is undefined or they are different
+                if (!this.matchRequestedClaimsHash(entity, requestedClaimsHash)) {
```

Comment:
Does this need a helper?
```suggestion
                if (entity.requestedClaimsHash !== requestedClaimsHash) {
```

Reply:
It doesn't really, just added it out of habit/consistency. Will apply your suggestion.